assets/js/components/legacy-notifications/util.js

```diff
@@ -87,10 +88,15 @@ const removeDismissed = ( notifications ) => {
 		return notifications;
 	}
 
-	return notifications.filter( ( notification ) => {
-		const dismissed = getCache( `notification::dismissed::${ notification.id }` );
+	const promises = await notifications.map( ( notification ) => getItem( `notification::dismissed::${ notification.id }` ) );
+	const allNotifications = await Promise.all( promises );
```

Comment:
Maybe something like `notificationDismissals` would be a bit more clear here since the value for each is only the cache entry for the notification's dismissed status.